src/state/mint/v3/utils.ts

```diff
@@ -7,9 +7,23 @@ import {
   TickMath,
 } from '@uniswap/v3-sdk/dist/'
 import { Price, Token } from '@uniswap/sdk-core'
-import { tryParseAmount } from 'state/swap/hooks'
 import JSBI from 'jsbi'
 
+export function tryParseAmountToPrice(baseToken?: Token, quoteToken?: Token, value?: string) {
+  if (!baseToken || !quoteToken || !value) {
+    return undefined
+  }
+
+  const numDecimals = value.indexOf('.') > -1 ? value.length - value.indexOf('.') - 1 : 0
+
+  return new Price(
+    baseToken,
+    quoteToken,
+    10 ** numDecimals * 10 ** baseToken.decimals,
+    parseFloat(value) * 10 ** numDecimals * 10 ** quoteToken.decimals
```

Comment:
is there a better way than indexOf + parseFloat? like can you get the number of digits after the decimal from the number instance?

Reply:
I think that `'.split('.')` is the simplest way to get the # of decimals.